example/flutter_example/lib/main.dart

```diff
@@ -38,16 +62,26 @@ class _HomePageState extends State<HomePage> {
                 crossAxisAlignment: CrossAxisAlignment.center,
                 mainAxisAlignment: MainAxisAlignment.center,
                 children: <Widget>[
+                  StreamBuilder(
+                    stream: chatter.subscription,
+                    builder: (context2, snapshot2) {
+                      if (snapshot2.hasData) {
+                        return Text('${snapshot2.data['msg']}');
+                      } else {
+                        return CircularProgressIndicator();
+                      }
+                    },
+                  ),
                   ActionChip(
-                    label: Text(snapshot.data == Status.CONNECTED
-                        ? 'DISCONNECT'
-                        : 'CONNECT'),
+                    label: Text(snapshot.data == Status.CONNECTED ? 'DISCONNECT' : 'CONNECT'),
                     backgroundColor: snapshot.data == Status.CONNECTED ? Colors.green[300] : Colors.grey[300],
                     onPressed: () {
-                      if (snapshot.data == Status.CONNECTED)
-                        ros.close();
-                      else
-                        ros.connect();
+                      print(snapshot.data);
+                      if (snapshot.data == Status.CONNECTED) {
```

Comment:
This should probably be something like `if (snapshot.data != Status.CONNECTED)`.

Reply:
Will fix